### ISSUE TITLE
Use ModelAdmin's Queryset for Searching

### ISSUE THREAD

**Issue by maqnius** [ahmedaljawahiry/django-admin-site-search#33]
Thank you for this project. It works great so far!

In one of my projects, we are using a lot of customized base querysets (ie. overriding `get_queryset`) in the ModelAdmins. Eg. for filtering the objects based on permissions.

It would be very useful, if the queryset for searching would not be [hardcoded to `objects.all()`](https://github.com/ahmedaljawahiry/django-admin-site-search/blob/main/admin_site_search/views.py#L180) but instead uses the ModelAdmin `get_queryset` method.

I would offer to create an according PR :)

Thanks!

**Comment by ahmedaljawahiry** [comment]
Hi! Glad you like the project 🙌 

Thanks for raising the issue - the hardcoded `objects.all()` is quite rigid. To keep it as customisable as possible, my instinct is to add a `self.get_model_queryset(self, request, model_class, model_admin)` method. Then, overriding that to use the admin's `get_queryset`, or whatever else, should be easy enough.

Feel free to raise a PR, otherwise I'll look to get something out asap 🙏

**Comment by maqnius** [comment]
@ahmedaljawahiry I see you already created a PR. Thank you!

I wanted to prepare a PR a few days ago but had a hard time getting the tests running locally. It would probably make sense to write a section about that in the readme, if you want others to contribute :)

**Comment by ahmedaljawahiry** [comment]
Thanks @maqnius - indeed lacking a "development" section in the readme, I'll look at getting this added.

I've merged that PR, and the `get_model_queryset(...)` method should be available in 1.1.0: https://github.com/ahmedaljawahiry/django-admin-site-search/releases/tag/1.1.0

To use the `ModelAdmin`'s method, I'd do something like:

```
class CustomAdminSite(AdminSiteSearchView, admin.AdminSite):
    """Adds the AdminSiteSearchView to the default AdminSite"""

    def get_model_queryset(
        self, request: HttpRequest, model_class: Model, model_admin: Optional[ModelAdmin],
    ) -> QuerySet:
        if model_admin:
            return model_admin.get_queryset(request)

        return super().get_model_queryset(request, model_class, model_admin)
```

Let me know if there's any issues 🙏
